morebits.js

```diff
@@ -3035,11 +3046,22 @@ Morebits.wiki.page = function(pageName, currentAction) {
 		if ($(xml).find('page').attr('title')) {
 			var resolvedName = $(xml).find('page').attr('title');
 
-			// only notify user for redirects, not normalization
 			if ($(xml).find('redirects').length > 0) {
+				// check for cross-namespace redirect:
+				var origNs = new mw.Title(ctx.pageName).namespace;
+				var newNs = new mw.Title(resolvedName).namespace;
+				if (origNs !== newNs && !ctx.followCrossNsRedirect) {
+					ctx.statusElement.error('Aborted: ' + ctx.pageName + ' is a cross-namespace redirect to ' + resolvedName);
```

Comment:
```suggestion
					ctx.statusElement.error('ctx.pageName + ' is a cross-namespace redirect to ' + resolvedName + ', aborted edit');
```

I think something like this, closer to the original, would be better; otherwise, you get a double colon, e.g. `Notifying initial contributor (<username>): Aborted: User talk:<pageName> is a cross-namespace redirect to <resolvedName>`